### ISSUE TITLE
heroku open -> Error opening web browser.

### ISSUE THREAD

**Issue by dlwagner** [dlwagner/issue-tracker#5]
Heroku open should open the web browser to the app login screen. It gives the following error instead:

`$ heroku open`
 `▸    Error opening web browser.`
` ▸    Error: Exited with code 3`
  `▸    Manually visit https://mysterious-escarpment-61469.herokuapp.com/ in your browser`

**Comment by dlwagner** [comment]
at least part of the problem lies with the fact I'm running these commands in VSCode terminal on WSL. This is probably not an issue with Heroku, but more an issue with properly configuring WSL.

**Comment by yedu-kn** [comment]
I also got the same error on WSL.

**Comment by luisgonzalex** [comment]
I am also having this issue even after doing the following in ~/.bashrc
`export BROWSER="/mnt/c/Program\ Files\ \(x86\)/Google/Chrome/Application/chrome.exe"`

**Comment by dotts-h** [comment]
`export BROWSER='/mnt/c/Program Files/Mozilla Firefox/firefox.exe` did the trick for me in WSL.

**Comment by marovargovcik** [comment]
Setting BROWSER variable as path to browser did not solve the issue for me. https://github.com/wslutilities/wslu has `wslview` utility that basically auto-picks default Windows browser and act as executable.

`wslview www.google.com` opens www.google.com with default browser in Windows
`wslview --register` or `wslview -r` will auto-register wslview as default browser in your distro (I believe it simply does this `update-alternatives set x-www-browser /usr/bin/wslview`)

So basically what worked for me is:
1. exporting BROWSER variable as `export BROWSER=wslview`
2. or using `wslview -r` to update distro defaults

I went with 2. as not adding more things to my `.bashrc` is preferable!

**Comment by musakhan17** [comment]
> Setting BROWSER variable as path to browser did not solve the issue for me. https://github.com/wslutilities/wslu has `wslview` utility that basically auto-picks default Windows browser and act as executable.
> 
> `wslview www.google.com` opens [www.google.com](http://www.google.com) with default browser in Windows
> `wslview --register` or `wslview -r` will auto-register wslview as default browser in your distro (I believe it simply does this `update-alternatives set x-www-browser /usr/bin/wslview`)
> 
> So basically what worked for me is:
> 
> 1. exporting BROWSER variable as `export BROWSER=wslview`
> 2. or using `wslview -r` to update distro defaults
> 
> I went with 2. as not adding more things to my `.bashrc` is preferable!

Could you explain a little step-by-step process as to how you found a fix for this? I'm using a bash terminal in Windows

**Comment by marovargovcik** [comment]
> > Setting BROWSER variable as path to browser did not solve the issue for me. https://github.com/wslutilities/wslu has `wslview` utility that basically auto-picks default Windows browser and act as executable.
> > `wslview www.google.com` opens [www.google.com](http://www.google.com) with default browser in Windows
> > `wslview --register` or `wslview -r` will auto-register wslview as default browser in your distro (I believe it simply does this `update-alternatives set x-www-browser /usr/bin/wslview`)
> > So basically what worked for me is:
> > 
> > 1. exporting BROWSER variable as `export BROWSER=wslview`
> > 2. or using `wslview -r` to update distro defaults
> > 
> > I went with 2. as not adding more things to my `.bashrc` is preferable!
> 
> Could you explain a little step-by-step process as to how you found a fix for this? I'm using a bash terminal in Windows

I found [this discussion](https://superuser.com/questions/1262977/open-browser-in-host-system-from-windows-subsystem-for-linux) where the second answer mentioned wslu. Following installation directions, it is mentioned that Ubuntu from the Windows Store has this installed by default and I looked up at the docs of wslview and found out about wslview -r command. Then looked it up in the source code to see what it does.  Hope that helps.

**Comment by ener1** [comment]
@marovargovcik thx~

**Comment by manoharthakur351** [comment]
me too having same error
````
▸    Error opening web browser.
 ▸    Error: Exited with code 3
 ▸
 ▸    Manually visit https://betabrains.herokuapp.com/ in your
 ▸    browser.
````

**Comment by Rafinha291-1** [comment]
> 2\. wslview -r

thanks bro, it really helped me with those, i didn't know what was wrong but i got that error, and it just worked fine, but everytime I start the aplication I have to run those on the wsl2 terminal otherwise it won't open automatically with heroku open command, help?
